internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -64,6 +64,69 @@ func computeCostComponent(region, machineType string, purchaseOption string, ins
 	}
 }
 
+// customComputeCostComponent returns a cost component for custom Compute instance usage.
+func customComputeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) []*schema.CostComponent {
+	sustainedUseDiscount := 0.0
+	if strings.ToLower(purchaseOption) == "on_demand" {
+		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
+		case "c2", "n2", "n2d":
+			sustainedUseDiscount = 0.2
+		case "n1", "f1", "g1", "m1":
+			sustainedUseDiscount = 0.3
+		}
+	}
+
+	qty := decimal.NewFromFloat(730)
+	if monthlyHours != nil {
+		qty = decimal.NewFromFloat(*monthlyHours)
+	}
+
+	cpuCostComponent := &schema.CostComponent{
+		Name:                fmt.Sprintf("Custom Instance CPU usage (Linux/UNIX, %s, %s)", purchaseOptionLabel(purchaseOption), machineType),
+		Unit:                "hours",
+		UnitMultiplier:      decimal.NewFromInt(1),
+		MonthlyQuantity:     decimalPtr(qty.Mul(decimal.NewFromInt(instanceCount))),
+		MonthlyDiscountPerc: sustainedUseDiscount,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr(region),
+			Service:       strPtr("Compute Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", ValueRegex: regexPtr("Custom Instance Core.*")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr(purchaseOption),
+		},
+	}
+
+	ramCostComponent := &schema.CostComponent{
+		Name:                fmt.Sprintf("Custom Instance RAM usage (Linux/UNIX, %s, %s)", purchaseOptionLabel(purchaseOption), machineType),
+		Unit:                "hours",
+		UnitMultiplier:      decimal.NewFromInt(1),
+		MonthlyQuantity:     decimalPtr(qty.Mul(decimal.NewFromInt(instanceCount))),
+		MonthlyDiscountPerc: sustainedUseDiscount,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr(region),
+			Service:       strPtr("Compute Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", ValueRegex: regexPtr("Custom Instance Ram.*")},
```

Comment:
@aliscott this set of attribute filters, when queried through the CLI, doesn't result in anything. But as a GraphQL query, they succeed.
```graphql
query {
  products(
    filter: {
      vendorName: "gcp",
      service: "Compute Engine",
      productFamily: "Compute",
      region: "europe-central2",
      attributeFilters: [
				{ key: "description", value_regex: "/Custom Instance Ram.*/" },
      ]
    },
  ) {
    attributes { key, value }
    prices { USD }
  }
}
```

Here's the terraform code example I'm using:
![image](https://user-images.githubusercontent.com/11949221/196426775-a0ba0b57-2b19-415f-a545-80c3a4a97fee.png)

Reply:
Hmm, looks like the `PurchaseOption` comes through differently from the Google prices for this resource.

This graphql query shows it as `OnDemand` instead of `on_demand`:

```graphql
query {
  products(
    filter: {
      vendorName: "gcp",
      service: "Compute Engine",
      productFamily: "Compute",
      region: "europe-central2",
      attributeFilters: [
				{ key: "description", value_regex: "/Custom Instance Ram.*/" },
      ]
    },
  ) {
    attributes { key, value }
    prices { USD, purchaseOption }
  }
}
```

Reply:
Not really! Thanks for the assistance! I'll try this asap.

Reply:
Got some progress here after these changes: https://github.com/infracost/infracost/pull/2004/commits/89ad55ff67d40b202698537a497d86b6d3b7ade8

I missed the fact that there are no preemptible custom instances. Only OnDemand.

My latest results are:
```bash
make run ARGS="breakdown --path ~/l/infracost-gcp-custom-instances/terraform"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version='" github.com/infracost/infracost/cmd/infracost breakdown --path ~/l/infracost-gcp-custom-instances/terraform
...
Evaluating Terraform directory at /home/tom/l/infracost-gcp-custom-instances/terraform
  ✔ Downloading Terraform modules
  ✔ Evaluating Terraform directory
  ✔ Retrieving cloud prices to calculate costs

Project: /home/tom/l/infracost-gcp-custom-instances/terraform

 Name                                                                   Monthly Qty  Unit           Monthly Cost
                                                                                                         
 google_container_node_pool.dev-1                                                                        
 ├─ Standard provisioned storage (pd-standard)                                  300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, custom-20-20480)        2,190  hours             $1,751.69
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, custom-20-20480)        2,190  gibibyte hour       $238.84
                                                                                                         
 OVERALL TOTAL                                                                                         $2,004.93

```

@aliscott now I'll figure out how to handle the corner case for custom instances with extended RAM.

Reply:
Got some progress here after these changes: https://github.com/infracost/infracost/pull/2004/commits/89ad55ff67d40b202698537a497d86b6d3b7ade8

I missed the fact that there are no preemptible custom instances. Only OnDemand.

My latest results are:
```bash
make run ARGS="breakdown --path ~/l/infracost-gcp-custom-instances/terraform"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version='" github.com/infracost/infracost/cmd/infracost breakdown --path ~/l/infracost-gcp-custom-instances/terraform
...
Evaluating Terraform directory at /home/tom/l/infracost-gcp-custom-instances/terraform
  ✔ Downloading Terraform modules
  ✔ Evaluating Terraform directory
  ✔ Retrieving cloud prices to calculate costs

Project: /home/tom/l/infracost-gcp-custom-instances/terraform

 Name                                                                   Monthly Qty  Unit           Monthly Cost
                                                                                                         
 google_container_node_pool.dev-1                                                                        
 ├─ Standard provisioned storage (pd-standard)                                  300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, custom-20-20480)        2,190  hours             $1,751.69
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, custom-20-20480)        2,190  gibibyte hour       $238.84
                                                                                                         
 OVERALL TOTAL                                                                                         $2,004.93

```

@aliscott now I'll figure out how to handle the corner case for custom instances with extended RAM.